src/pretix/plugins/sendmail/models.py

```diff
@@ -0,0 +1,400 @@
+                return get_email_context(
+                    event=self.event,
+                    order=order,
+                    position=position,
+                    position_or_address=position_or_address,
+                )
+            else:
+                return get_email_context(event=self.event)
+        except Exception as e:
+            logger.exception("Error while generating email context")
+            recipient.error = f"Context error: {str(e)}"
+            recipient.save(update_fields=["error"])
+            return None
+
+    def _finalize_send_status(self):
+        if all(r.sent for r in self.recipients.all()):
+            self.sent_at = now()
+        else:
+            self.sent_at = None
+        self.save(update_fields=["sent_at"])
+
+    def _send_to_recipient(self, recipient, subject, message):
+        from pretix.base.services.mail import SendMailException
+        email = recipient.email
+        if not email:
+            return False
+
+        order_id = recipient.orders[0] if recipient.orders else None
+        position_id = recipient.positions[0] if recipient.positions else None
+
+        order = Order.objects.filter(pk=order_id, event=self.event).first() if order_id else None
+        position = OrderPosition.objects.filter(pk=position_id).first() if position_id else None
+
+        try:
+            ia = order.invoice_address if order else None
+        except InvoiceAddress.DoesNotExist:
+            ia = InvoiceAddress(order=order) if order else None
+
+        position_or_address = position or ia
+        context = self._build_email_context(order, position, position_or_address, recipient)
+        if context is None:
+            return True  # Error already logged
+
+        try:
+            mail(
+                email=email,
+                subject=subject,
+                template=message,
+                context=context,
+                event=self.event,
+                locale=order.locale if order else self.locale,
+                order=order,
+                position=position,
+                sender=self.event.settings.get('mail_from'),
+                event_bcc=self.bcc,
+                event_reply_to=self.reply_to,
+                attach_cached_files=self.attachments,
+                user=self.user,
+                auto_email=False,
+            )
+            recipient.sent = True
+            recipient.error = None
+        except SendMailException as se:
+            recipient.sent = False
+            recipient.error = str(se)
+            recipient.save(update_fields=["sent", "error"])
+            logger.exception("SendMailException error while sending to %s: %s", email, str(e))
+        except Exception as e:
+            recipient.sent = False
+            recipient.error = f"Internal error: {str(e)}"
+            recipient.save(update_fields=["sent", "error"])
+            logger.exception("Unexpected error while sending to %s: %s", email, str(e))
```

Comment:
When using `log.exception`, you don't need to embed error to the logging message.
The error will be run with traceback separately.